src/test/kotlin/de/bund/digitalservice/useid/widget/WidgetControllerIntegrationTest.kt

```diff
@@ -116,8 +113,9 @@ class WidgetControllerIntegrationTest(@Autowired val webTestClient: WebTestClien
         val iOSResponseBody = iosResponse.expectBody().returnResult().responseBody?.decodeToString()
         val androidResponseBody = androidResponse.expectBody().returnResult().responseBody?.decodeToString()
 
-        assertThat(iOSResponseBody, containsString(widgetProperties.errorView.incompatible.localization.headlineTitle))
-        assertThat(androidResponseBody, containsString(widgetProperties.errorView.incompatible.localization.headlineTitle))
+        val errorTitle = "Ihr Smartphone erfüllt leider nicht die Voraussetzungen der BundesIdent App."
```

Comment:
How does the test suite know that we are testing in `de` locale? This would break the test if we run in different locale, right? or maybe we should not test against string... 🤔

Reply:
+1

Reply:
German is the default (`message.properties`) if no locale is set. Currently the locale is resolved by checking the `Accept-Language` HTTP header (default behaviour). As long as we don't set this header for the test (or if the header includes locales which are not supported) the language will default to german which is why this test succeeds. 
In my opinion we can leave that for now and once we support multiple languages, we add tests which set the language header and validate the language of the widget accordingly. 
Other option would be to replace the hardcoded String with reading the value from the `MessageSource`.
What do you think?

Reply:
I agree, that both ways have pros and cons, that's always a problem. What we actually wanted would be that all the used keys have a non-empty value, but that's rather another test, so let's use MessageSource here.